src/Multi_filtration/include/gudhi/One_critical_filtration.h

```diff
@@ -39,266 +42,477 @@ namespace Gudhi::multi_filtration {
+
+    One_critical_filtration result(f.size(), value);
+    result /= f;
+    return result;
   }
 
-  // template<class array_like>
-  inline friend bool operator==(const One_critical_filtration<T> &self, const One_critical_filtration<T> &to_compare) {
-    if (self.num_parameters() != to_compare.num_parameters()) return false;
-    auto it = to_compare.begin();
-    for (auto i = 0u; i < self.num_parameters(); i++) {
-      if (self.at(i) != *(it++)) return false;
+  friend One_critical_filtration &operator/=(One_critical_filtration &result, const One_critical_filtration &to_div)
+  {
+    if (result.empty()) return result;
+
+    bool res_is_infinite = result.is_inf() || result.is_minus_inf();
+    bool to_div_is_infinite = to_div.is_inf() || to_div.is_minus_inf();
+
+    if (result.is_nan() || to_div.is_nan() || (res_is_infinite && to_div_is_infinite)) {
+      result = nan();
+      return result;
+    }
+
+    if (to_div_is_infinite) {
+      return apply_scalar_operation_on_finite_value_with_all_nan_possible_(result, to_div[0], divide_);
     }
-    return true;
-  }
 
-  inline static std::vector<std::vector<T>> to_python(const std::vector<One_critical_filtration<T>> &to_convert) {
-    return std::vector<std::vector<T>>(to_convert.begin(), to_convert.end());
+    GUDHI_CHECK(res_is_infinite || result.size() == to_div.size(),
+                "Two filtration points with different number of parameters cannot be divided.");
+
+    if (res_is_infinite) {
+      result.resize(to_div.size(), result[0]);
+    }
+
+    return apply_operation_with_finite_values_(result, to_div, divide_);
   }
 
-  inline static std::vector<One_critical_filtration<T>> from_python(const std::vector<std::vector<T>> &to_convert) {
-    return std::vector<One_critical_filtration<T>>(to_convert.begin(), to_convert.end());
+  friend One_critical_filtration &operator/=(One_critical_filtration &result, const T &to_div)
+  {
+    if (result.empty()) return result;
+
+    bool res_is_infinite = result.is_inf() || result.is_minus_inf();
+    bool to_div_is_infinite = to_div == T_inf || to_div == -T_inf;
+
+    if (to_div == 0 || std::isnan(to_div) || result.is_nan() || (res_is_infinite && to_div_is_infinite)) {
+      result = nan();
+      return result;
+    }
+
+    if (res_is_infinite) {
+      if (to_div < 0) result[0] = -result[0];
+      return result;
+    }
+
+    return apply_scalar_operation_on_finite_value_with_all_nan_possible_(result, to_div, divide_);
   }
 
+  // MODIFIERS
+
   /** \brief This functions take the filtration value `this` and pushes it to
    * the cone \f$ \{ y\in \mathbb R^n : y>=x \} \f$. After calling this method,
    * the value of this is updated to \f$ \mathrm{this} = \min \{ y\in \mathbb
    * R^n : y>=this \}\cap \{ y\in \mathbb R^n : y>=x \}
    * @param[in] x The target filtration value on which to push `this`.
    */
-  inline void push_to(const One_critical_filtration<T> &x) {
+  void push_to(const One_critical_filtration &x)
```

Comment:
We could maybe rename this to `push_to_common_majorant` to be coherent with other methods,
but that's not very important

Reply:
Coherent with which other method?

Reply:
I don't know exactly where in the original PR, but we discussed with Marc to rename the push to `least_common_majorant` somewhere, it's not done yet indeed.

Reply:
`push_to_least_common_majorant` and `pull_to_greatest_common_minorant` sounds fine to me. Should I change it?